blob/blob.go

```diff
@@ -101,41 +102,48 @@ type Blob struct {
 	types.Blob `json:"blob"`
 
 	Commitment Commitment `json:"commitment"`
+
+	// the celestia-node's namespace type
+	// this is to avoid converting to and from app's type
+	namespace share.Namespace
 }
 
-// NewBlob constructs a new blob from the provided namespace.ID and data.
-func NewBlob(shareVersion uint8, namespace namespace.ID, data []byte) (*Blob, error) {
-	if len(namespace) != appns.NamespaceSize {
-		return nil, fmt.Errorf("invalid size of the namespace id. got:%d, want:%d", len(namespace), appns.NamespaceSize)
-	}
+// NewBlobV0 constructs a new blob from the provided Namespace and data.
+// The blob will be formatted as v0 shares.
+func NewBlobV0(namespace share.Namespace, data []byte) (*Blob, error) {
```

Comment:
`NewBlobV0`, `NewBlobV1`, .... `NewBlobVn`. Why can't we make NewDefaultBlob where we can use `DefaultShareVersion` ? And one more ctor that takes a specific version

Reply:
n/a. Agreed in DM that having a Default ctor will break the user's soft after moving to the new default version